Example-iOS-Swift/WebViewController.swift

```diff
@@ -78,16 +79,25 @@ class WebViewController: UIViewController, WKScriptMessageHandler {
     }
     
     func userContentController(_ userContentController: WKUserContentController, didReceive message: WKScriptMessage) {
-        
         if let body = message.body as? String{
-            
             let jsonData = body.data(using: .utf8)
             
-            if let bodyStruct = try? JSONDecoder().decode(MessageData.self, from: jsonData!){
+            if let bodyStruct = try? JSONDecoder().decode(MessageData.self, from: jsonData!) {
                 switch bodyStruct.eventName {
                 case "v1.avatar.exported":
-                    avatarUrlDelegate?.avatarUrlCallback(url : bodyStruct.data!["url"]!)
+                    let event = AvatarExportedEvent (url: bodyStruct.data!["url"]!)
+                    webViewDelegate?.onAvatarExported(event: event)
                     reloadPage(clearHistory: false)
+                case "v1.asset.unlock":
+                    let event = AssetUnlockedEvent (userId: bodyStruct.data!["userId"]!,
+                                                    assetId: bodyStruct.data!["assetId"]!)
+                    webViewDelegate?.onAssetUnlocked(event: event)
+                case "v1.user.set":
+                    let event = UserSetEvent (id: bodyStruct.data!["id"]!)
+                    webViewDelegate?.onUserSet(event: event)
+                case "v1.user.authorized":
+                    let event = UserAuthorizedEvent (url: bodyStruct.data!["url"]!)
```

Comment:
THis causes error because url doesn't exist, I'm guessing its just a copy pasta issue, it should be "id" instead. 
Another thing to perhaps consider here is maybe handling if the key doesn't exist, at the moment if the body data is missing this field it will throw and error and stop the application.

Reply:
I actually took this from the react avatar creator a bit weird, which why I've also check with player experience to get a full list of the possible events. Will change this here, and also update the react avatar creator too as I assume that is wrong too 😅

Reply:
Fixed 👍